File: Intermediate Algorithm Scripting/Javascript/convertHTML.js
/*
Convert the characters &, <, >, " (double quote), and ' (apostrophe), in a string to their corresponding HTML entities.
*/

function convertHTML(str) {
  let convObj = {
    "&": "&amp;",
    "<": "&lt;",
    ">": "&gt;",
    '"': "&quot;",
    "'": "&apos;",
  };
  return str.replace(/[&<>"']/g, function (match) {
    return convObj[match];
  });
}

File: Intermediate Algorithm Scripting/Python/convertHTML.py
# Convert the characters &, <, >, " (double quote), and ' (apostrophe), in a string to their corresponding HTML entities.


def convertHTML(string: str):
    convDict = {"&": "&amp;", "<": "&lt;", ">": "&gt;", '"': "&quot;", "'": "&apos;"}
    for x in string:
        if x in convDict.keys():
            string = string.replace(x, convDict[x])
    return string
